src/test/scala/com/target/data_validator/validator/RowBasedSpec.scala

```diff
@@ -115,10 +115,53 @@ class RowBasedSpec extends FunSpec with Matchers with TestingSparkSession {
       val dict = new VarSubstitution
       val schema = StructType(List(StructField("d", DoubleType)))
       val df = spark.createDataFrame(sc.parallelize(List(Row(-1.0), Row(0.0), Row(1.0))), schema)
-      val sut = mkConfig(df, NegativeCheck("d") :: Nil)
+      val sut = mkConfig(df, NegativeCheck("d", None) :: Nil)
       assert(sut.quickChecks(spark, dict))
     }
 
   }
 
+  describe ("threshold parsing") {
+    describe ("validate different way of specifying thresholds") {
+      it ("absolute 10") {
+        val sut = NullCheck("col", Some("10"))
+        assert(!sut.configCheckThreshold)
+      }
+
+      it ("less then 1.0") {
+        val sut = NullCheck("col", Some("0.10"))
+        assert(!sut.configCheckThreshold)
+      }
+
+      it ("100%") {
+        val sut = NullCheck("col", Some("10%"))
+        assert(!sut.configCheckThreshold)
+      }
+
+      it("bad value") {
+        val sut = NullCheck("col", Some("peanuts"))
+        assert(sut.configCheckThreshold)
+        assert(sut.failed)
+      }
```

Comment:
I think we need to handle negative cases.  The question is how?  Do we error out, or just treat them as a threshold of 0?

```scala
      it ("is negative") {
        val sut = NullCheck("col", Some("-10"))
        ???
      }
      it ("is negative fraction") {
        val sut = NullCheck("col", Some("-0.1"))
        ???
      }
      it ("is negative percent") {
        val sut = NullCheck("col", Some("-10%"))
        ???
      }
```

Reply:
I agree, added these and a couple more.